src/std/net/socket/kqueue-server.ss

```diff
@@ -4,41 +4,54 @@
 package: std/net/socket
 
 (require bsd)
+
 (import :gerbil/gambit/threads
         :gerbil/gambit/ports
         :gerbil/gambit/misc
         :std/net/socket/base
         :std/net/socket/basic-server
         :std/os/fd
         :std/os/kqueue
+        :std/os/error
         :std/iter)
+
 (export kqueue-socket-server)
 
+(declare
+  (fixnum))
+
 (def (kqueue-socket-server)
   (def fdtab (make-hash-table-eq))
   (def kq (kqueue))
   (def maxevts 8192)
   (def evts (make-kevents maxevts))
-
-  (def ev-in (##fxior EVFILT_READ EV_EOF EV_ERROR))
-  (def ev-out (##fxior EVFILT_WRITE EV_EOF EV_ERROR))
```

Comment:
we still want event notifications for EOF, where is this handled?

Reply:
In each of the three cases, data available (or data can be written), eof, and error, an `EVFILT_READ` (or `EVFILT_WRITE`) event is generated for the file descriptor. We could distinguish between the three cases by examining the flags field of the event. But this is not necessary, as all three cases have to be handled in the same way. So the tests for `EVFILT_READ` and `EVFILT_WRITE` also include the eof case.

And I realize that I should remove also the test for `EV_ERROR` to get the same behavior as in the implementation with epoll.